frontend/src/components/VCollectionHeader/VCollectionHeader.vue

```diff
@@ -0,0 +1,143 @@
+<template>
+  <div
+    class="collection-header grid grid-cols-[1.5rem,1fr] gap-x-2 gap-y-4 md:grid-rows-[auto,auto] md:gap-y-8"
+    :class="
+      collection === 'tag'
+        ? 'tags grid-rows-[auto,auto] md:grid-cols-[2.5rem,1fr]'
+        : 'grid-rows-[auto,auto,auto] md:grid-cols-[2.5rem,1fr,auto]'
+    "
+  >
+    <VIcon :name="iconName" :size="isMd ? 10 : 6" class="icon" />
+    <h1 class="title text-3xl font-semibold leading-snug md:text-6xl">
+      {{ title }}
+    </h1>
+    <VButton
+      v-if="collection !== 'tag' && url"
+      as="VLink"
+      variant="filled-dark"
+      size="medium"
+      class="button label-bold !flex-none md:ms-4"
+      has-icon-end
+      show-external-icon
+      :external-icon-size="6"
+      :href="url"
+      @click="sendAnalyticsEvent"
+      >{{ $t(`collection.link.${collection}`) }}</VButton
+    >
+    <p
+      class="results caption-regular md:label-regular mt-2 text-dark-charcoal-70 md:mt-0"
+    >
+      {{ resultsLabel }}
+    </p>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, PropType } from "vue"
+
+import { useRoute } from "@nuxtjs/composition-api"
+
+import { useUiStore } from "~/stores/ui"
+import type { Collection } from "~/types/search"
+import { useAnalytics } from "~/composables/use-analytics"
+
+import VButton from "~/components/VButton.vue"
+import VIcon from "~/components/VIcon/VIcon.vue"
+
+const icons = {
+  tag: "tag",
+  source: "institution",
+  creator: "person",
+}
+
+/**
+ * Renders the header of a tag/creator/source collection page.
+ */
+export default defineComponent({
+  name: "VCollectionHeader",
+  components: { VIcon, VButton },
+  props: {
+    collection: {
+      type: String as PropType<Collection>,
+      required: true,
+    },
+    /**
+     * The name of the tag/creator/source. The source name should be the display
+     * name, not the code.
+     */
+    title: {
+      type: String,
+      required: true,
+    },
+    url: {
+      type: String,
+    },
+    /**
+     * The label showing the result count, to display below the title.
+     */
+    resultsLabel: {
+      type: String,
+      required: true,
+    },
```

Comment:
Passing the result label manually like this results in the storybook example not quite matching the designs, but only be consequence of the story configuration. The creator design has different text than the other collections:

![image](https://github.com/WordPress/openverse/assets/24264157/78641a37-802d-48fc-a0b3-8fa86bd423e7)

If we changed the component to accept a configuration object on the `collection` prop where `collection` could be a union of the following interfaces:

```typescript
interface BaseCollection {
  type: unknown
  mediaType: MediaType
  resultCount: number
  title: string
}

interface TagCollection extends BaseCollection {
  type: 'tag'
}

interface SourceCollection extends BaseCollection {
  type: 'source'
  url: string
}

interface CreatorCollection extends BaseCollection {
  type: 'creator'
  source: string
  url: string
}
```

Then the component should select the correct template string and interpolate `title`, `resultCount`, `mediaType` and, if a creator, `source`.

There are probably some linguistic difficulties here though that might warrant passing individual source names to the translators... Some sources really need a definite article and others do not. Maybe leaving the definite article out is okay in English for some. "10000 results from Museum of Modern Art" reads clumsy to me. "10000 results from the Museum of Modern Art" would be better. But "10000 results from the NASA" is incorrect. In Portuguese the definite article is always included in the word "from" ("10000 resultados do Museum of Modern Art"). However, even then, the grammatical gender should be considered. If we had a source "Biblioteca Nacional do Brasil", the grammatical gender is feminine and it should be "10000 resultados da Biblioteca Nacional do Brasil".

I wonder if everywhere we are going to interpolate a source name into a sentence, whether we need to create a translation string for it, like automatically generating certain translations from the stats endpoints. We wouldn't translate the source names themselves, necessarily (though that may be appropriate in some cases, and transliterating source names could be useful for [languages with non-Latin script](https://ja.wikipedia.org/wiki/ニューヨーク近代美術館)), but it would still be necessary context for minimally correct translation in some languages. It would be jarring to read "10000 resultados do Public Library of New York" for an adult with English literacy, but all the more so (and potentially incomprehensible) for a child, e.g., in an educational context. We can't do anything about creator names, of course, but the list of sources is trivially obtainable and limited (unlike creator names of which there are probably hundreds of thousands and most cannot or shouldn't be translated or transliterated anyway).

If we're going down the road of creating sentences from source names, however, it does seem like something that we could consider in an entirely separate issue. I don't know exactly how adding those strings to `en.json5` would work, if we could do so reliably, etc. We could go with a naive approach now that avoids needing to do any translations... maybe "10000 images - Biblioteca Nacional do Brasil", "10000 tagged images", and "10000 images - sarayourfriend, Flickr" (not concrete suggestions).

In either case, I think passing the `resultsLabel` in this way sort of invites accidental misuse or unintentionally duplicating similar translation strings.

Reply:
Great catch and suggestions.

Translating source names sound interesting, we can certainly can create an improvement issue to provide better context. So far, I like the following suggestion for the sentence.

> 10000 images - sarayourfriend, Flickr

Although I would replace the dash with a period to end up with something like this: `10000 images. sarayourfriend, Flickr`

Reply:
@sarayourfriend, I started to implement changes to the label here, and then realized that it needs more props for some of the collection kinds (media type to show whether it's "... images" or "...audio files", results count, etc.). It's better to create the label in the parent component (the collection page that already has all the necessary information)

Reply:
That's why I suggested the expanded collection type, to encode the required information for each collection type. I generally have a preference for logic being as "low" in the component tree with higher components only responsible for overall layout, but where you think it would be best to do. Anywhere we put it will include the same complexity.